scripts/project-trigger.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+
+json_file="../projects/projects.json"
+gh_token=$GH_TOKEN
+git_ref="main"
+workflow_organization_name="cncf-tags"
+workflow_project_name="green-reviews-tooling"
+workflow_dispatcher_file_name="benchmark-pipeline.yaml"
+
+if [ -z "$gh_token" ]; then
+    echo "[ERR] GH_TOKEN not set"
+    exit 20
+fi
+
+jq -c '.projects[]' "$json_file" | while read -r project; do
+    proj_name=$(echo "$project" | jq -r '.name')
+    proj_organization=$(echo "$project" | jq -r '.organization')
+    sub_components=$(echo "$project" | jq -r '.sub_components')
+
+    echo "[DBG] Project Name: $proj_name"
+    echo "[DBG] Organization: $proj_organization"
+    echo "[DBG] SubComponents: $sub_components"
+
+    release_url="https://api.github.com/repos/${proj_organization}/${proj_name}/releases/latest"
+    
+    
+    response=$(curl --fail-with-body -sSL -X GET $release_url)
+    status_code=$?
+    if [ $status_code -ne 0 ]; then
+        echo "[ERR] fetching latest release for ${proj_name} from ${release_url}. Status code: $status_code"
+        echo "[ERR] Response: $response"
+        continue
+    fi
+
+    latest_proj_version=$(echo "$response" | jq -r '.tag_name')
+    if [ -z "$latest_proj_version" ]; then
+        echo "[ERR] Could not find the latest version for ${proj_name}"
+        continue
+    fi
+
+    echo "[DBG] Version: $latest_proj_version"
+
+    # TODO(dipankar): need to make the cncf_project_sub to be each individiual call for each sub-component
```

Comment:
Yes, this still needs to be implemented.

Reply:
> also need to think of do we send one sub_component at a time or all at once like a json array or a string with each having its individual workflow_call?

The benchmark-pipeline action needs to be called once per subcomponent if they exist.

Its an array in the projects.json so can you loop through each component? 

If there are no sub components you could pass an empty string.